monkey/common/cloud/aws/aws_instance.py

```diff
@@ -15,36 +18,54 @@
 AWS_TIMEOUT = 2
 
 
+@dataclass
+class AwsInstanceInfo:
+    instance_id: Optional[str] = None
+    region: Optional[str] = None
+    account_id: Optional[str] = None
+
+
 class AwsInstance(CloudInstance):
     """
     Class which gives useful information about the current instance you're on.
     """
 
-    def is_instance(self):
-        return self.instance_id is not None
+    __metaclass__ = Singleton
 
     def __init__(self):
-        self.instance_id = None
-        self.region = None
-        self.account_id = None
+        self._is_instance, instance_info = AwsInstance._fetch_instance_info()
+
+        self.instance_id = instance_info.instance_id
+        self.region = instance_info.region
+        self.account_id = instance_info.account_id
```

Comment:
If you make these properties, they become immutable from outside the class, which could be desirable, especially since this class is a singleton.

Reply:
I don't think this is a well communicated immutability mechanism. I'll change these to private though

Reply:
I'm not sure I follow. If you make them properties without setters, they can't be changed from outside the class (without jumping through some hoops).